libs/langgraph/langgraph/pregel/__init__.py

```diff
@@ -676,30 +677,60 @@ def update_state(
         self,
         config: RunnableConfig,
         values: Optional[Union[dict[str, Any], Any]],
-        as_node: Optional[str] = None,
+        as_node: Optional[str | list[str]] = None,
     ) -> RunnableConfig:
         """Update the state of the graph with the given values, as if they came from
         node `as_node`. If `as_node` is not provided, it will be set to the last node
         that updated the state, if not ambiguous.
         """
-        if not self.checkpointer:
+        checkpointer: Optional[BaseCheckpointSaver] = config["configurable"].get(
+            CONFIG_KEY_CHECKPOINTER, self.checkpointer
+        )
+        if not checkpointer:
             raise ValueError("No checkpointer set")
 
+        has_nested_interrupts = _has_nested_interrupts(self)
+        if (
+            has_nested_interrupts
+            and as_node is not None
+            and not isinstance(as_node, list)
+        ):
+            raise ValueError(
```

Comment:
what is this for? we should accept string as having the same meaning as a list of length 1